Search status filter

Timo614 opened the pull request:
Fixes: #270 

This branch addresses the associated issue by adding a new status filter that takes a comma separated list.

Notes:
- Fixes the Event create logic to use the NewEvent struct
- NewEvent doesn't deserialize the status field (which is why I was guessing the other struct was used?) now and instead uses `Draft` by default (using serde field attributes) -- this also fixed an issue where some of the fields were accepted but not settable given the old struct didn't pass them along
- Event search now doesn't return `Draft` events
- Didn't modify the event search related indexes -- it's something we'll likely want to do in the future before we launch along with other performance evaluations

Examples:

Event search with no status returns all three events (`Closed`, `Published`, and `Offline`)

```
curl http://localhost:9000/events
[{"id":"c0144881-8442-4e4d-9630-fed0564c5224","name":"ClosedEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:05:58.827900","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Closed","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null},{"id":"69126d96-bad9-44bf-9c0d-43e1ce021612","name":"OfflineEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:06:14.204014","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Offline","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null},{"id":"11183296-d102-416d-a301-e4d1c3797ef9","name":"PublishedEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:05:53.068440","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Published","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null}]
```

Event search with `Closed` status alone:

```
curl http://localhost:9000/events?status=Closed
[{"id":"c0144881-8442-4e4d-9630-fed0564c5224","name":"ClosedEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:05:58.827900","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Closed","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null}]
```

Event search with `Closed` and `Published` statuses:

```
curl http://localhost:9000/events?status=Published,Closed
[{"id":"c0144881-8442-4e4d-9630-fed0564c5224","name":"ClosedEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:05:58.827900","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Closed","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null},{"id":"11183296-d102-416d-a301-e4d1c3797ef9","name":"PublishedEvent","organization_id":"c5025c48-0e2b-4e6a-9452-f015690bb50b","venue_id":null,"created_at":"2018-09-25T10:05:53.068440","event_start":"2018-09-24T15:05:40.859024","door_time":null,"status":"Published","publish_date":null,"promo_image_url":null,"additional_info":null,"age_limit":null,"cancelled_at":null,"venue":null}]
```

mikethetike commented on db/src/models/enums.rs:
```diff
@@ -37,6 +37,7 @@ macro_rules! string_enum {
 
 string_enum! { AssetStatus [Unsynced] }
 string_enum! { EventStatus [Draft,Closed,Published,Offline]}
+string_enum! { EventSearchStatus [Cancelled, Live]}
```
I'd be careful about using `Live` as a status because it could be confused with a Live Event (music)

Timo614 replied:
Sounds good yeah "Active" may be better. It was mentioned in the associated issue we can just use EventStatus instead so will just switch the logic to that.

mikethetike commented on db/src/models/events.rs:
```diff
@@ -49,6 +50,19 @@ pub struct NewEvent {
+            .to_db_error(ErrorCode::InsertError, "Could not create new event")
+    }
+
+    pub fn default_status() -> String {
```
Clever

mikethetike commented on db/src/models/events.rs:
```diff
@@ -64,18 +78,10 @@ pub struct EventEditableAttributes {
 impl Event {
     pub fn create(
         name: &str,
+        status: &str,
```
If possible, I think it would be better to use the EventStatus enum when methods are called programmatically, but sometimes that is not practical, so I understand if this needs to be a string

Timo614 replied:
Agreed, will make this change.

Timo614 commented:
I've switched the logic to use a comma separated array of EventStatus values as per a comment in the associated issue and adjusted the logic for the review feedback.
